pyquil/quil.py

```diff
@@ -1014,6 +1173,7 @@ def merge_with_pauli_noise(
     return p
 
 
+# TODO: does this need modification?
```

Comment:
calibrations, waveforms, and frames need to be re-attached to the output program. Those may need merging themselves.

Reply:
Conflict in waveform or frame names should be an error. Calibrations with same spec may override earlier calibrations without error.

Reply:
This could be  combined with the `__add__` method to merge pairwise.

Reply:
Combined or replaced by? i.e.
```
def merge_programs(programs):
    return programs[0] + programs[1] + ... + programs[n-1]
```

Reply:
Yup, could replace it with a reduction